components/Panel/HistoryPanel.tsx

```diff
@@ -0,0 +1,140 @@
+import { useUserContext, useVotesContext } from "hooks";
+import styled, { CSSProperties } from "styled-components";
+import { VoteT } from "types";
+import { PanelFooter } from "./PanelFooter";
+import { PanelTitle } from "./PanelTitle";
+import { PanelSectionText, PanelSectionTitle, PanelWrapper } from "./styles";
+
+export function HistoryPanel() {
+  const { getPastVotes, getIsFetching } = useVotesContext();
+  const { apr, cumulativeCalculatedSlash, cumulativeCalculatedSlashPercentage, userDataFetching } = useUserContext();
+  const bonusPenaltyHighlightColor = cumulativeCalculatedSlashPercentage?.eq(0)
+    ? black
+    : cumulativeCalculatedSlashPercentage?.gt(0)
+    ? green
+    : red500;
+
+  function isLoading() {
+    return getIsFetching() || userDataFetching;
+  }
+
+  return (
+    <PanelWrapper>
+      <PanelTitle title="History" />
+      <SectionsWrapper>
+        <AprWrapper>
+          <AprHeader>Your return</AprHeader>
+          <Apr>
+            {isLoading() ? (
+              <LoadingSkeleton variant="white" width={150} height={35} />
+            ) : (
+              `${formatNumberForDisplay(apr)}%`
+            )}
+          </Apr>
+          <AprDetailsWrapper>
+            <Text>
+              <>
+                Based on participation score ={" "}
+                {isLoading() ? (
+                  <LoadingSkeleton width={60} height={15} />
+                ) : (
+                  formatNumberForDisplay(cumulativeCalculatedSlash)
+                )}
+              </>
+            </Text>
+            <Text>
+              Your bonus/penalty ={" "}
+              <BonusOrPenalty
+                style={
+                  {
+                    "--color": bonusPenaltyHighlightColor,
+                  } as CSSProperties
+                }
+              >
+                {isLoading() ? (
+                  <LoadingSkeleton width={60} height={15} />
+                ) : (
+                  `${formatNumberForDisplay(cumulativeCalculatedSlashPercentage)}%`
+                )}
+              </BonusOrPenalty>
+            </Text>
+          </AprDetailsWrapper>
+        </AprWrapper>
+        <SectionWrapper>
+          <PanelSectionText>
+            Lorem ipsum dolor sit amet consectetur adipisicing elit. Reprehenderit enim voluptate rem perferendis
+            numquam, consequuntur sapiente nesciunt laudantium quibusdam pariatur.
+          </PanelSectionText>
+        </SectionWrapper>
+        <SectionWrapper>
+          <PanelSectionTitle>Voting history</PanelSectionTitle>
+          <HistoryWrapper>
+            {isLoading() ? (
+              <LoadingSpinner size={250} />
+            ) : (
+              <VoteHistoryTable votes={getPastVotes().sort(sortVotesByVoteNumber)} />
+            )}
+          </HistoryWrapper>
+        </SectionWrapper>
+      </SectionsWrapper>
+      <PanelFooter />
+    </PanelWrapper>
+  );
+}
+
+function sortVotesByVoteNumber(a: VoteT, b: VoteT) {
```

Comment:
We could also sort by timestamp or something else, but I like how consecutive vote numbers look